source/architecture-partials/architecture.md

```diff
@@ -0,0 +1,100 @@
+## Architecture diagram
+![alt text](/images/eff-architecture.svg "High-level architecture of the Ethnicity Facts and Figures website and Publisher showing the relationships between components in AWS, Heroku and Github.")
+
+[View/edit source file for architecture diagram](https://docs.google.com/presentation/d/1lbRGSTxkQN9C7blpIPZq0dhgOrgrEVeKU5DJiJp_HQw/edit)
+
+## The Ethnicity Facts and Figures website
+
+The public-facing [Ethnicity Facts and Figures site](https://www.ethnicity-facts-figures.service.gov.uk/) is a set of 
+static pages in an S3 bucket (actually two S3 buckets - a primary one in London and secondary mirror in Ireland) which 
+are served through a Cloudfront distribution.  The staging static site is set up in the same way, but with an additional
+AWS Lambda function to provide basic auth (this is intended to prevent indexing by search engines).
+
+The [style guide](https://guide.ethnicity-facts-figures.service.gov.uk/) is a [GitHub Pages](https://pages.github.com/)
+site that is set up to serve from the same domain.
+
+DNS records for `ethnicity-facts-figures.service.gov.uk` are managed in Amazon Route 53.
+
+## The Publisher application
+
+The Publisher is a [Python](https://www.python.org/) web application based on the 
+[Flask web framework](http://flask.pocoo.org/) backed by a [Postgres](https://www.postgresql.org/) database.
+
+The Publisher app is hosted on Heroku and released through a three-stage pipeline:
+
+1. **Review app** When a pull request is opened on the Publisher in GitHub the tests are run and a review app is created
+   in Heroku.
+   Review apps get a database copied from the Staging database, but with at most two measure pages per subtopic
+   (review app databases are currently limited by Heroku to 10,000 rows, so we can't use a full copy).
+
+2. **Staging** When a pull request is merged into the `master` branch the tests are run again and the latest `master`
+   code is automatically deployed to the Staging environment.
+   Staging has a full copy of the production database, but with all user emails and passwords removed.  There is a set
+   of user accounts set up for each different role.
+
+3. **Production** Promoting code to production is a manual step: use the "Promote to production..." button in the 
+   Heroku pipeline.  This will immediately release the master branch code to production, with no further test run.
+
+## Static site build
+
+The Publisher app allows users to create, update, edit and preview pages for the Ethnicity Facts and Figures site. 
+For these changes to be reflected in the pages of the live site they need to be built and pushed to the production S3
+bucket. A build of the entire site is requested by Publisher whenever a page is marked as published (or, in rare cases,
```

Comment:
Also when new code is released, I think?

Reply:
I 100% meant to put that in.